docs/gcs.md

```diff
@@ -0,0 +1,160 @@
+
+```java
+Map<String, String> properties = new HashMap<>();
+properties.put("inputDataLocation", "gs://my_bucket/data/");
+properties.put("metadataLocation", "gs://my_bucket/metadata/");
+gcsFileIO.initialize(properties);
+```
+
+Within these property key-value pairs, inputDataLocation and metadataLocation are the locations in your GCS bucket where your data and metadata are stored. Update `"gs://my_bucket/data/" ` and `"gs://my_bucket/metadata/" ` to reflect the corresponding paths of your GCS bucket.
+
+### Example Use of GCSFileIO
+
+Once `GCSFileIO` is initialized and configured, you can interact with the data housed on GCS. Below, we will demonstrate how to create and access an `InputFile` and an `OutputFile`.
+
+- **Creating an InputFile**: To create an `InputFile` for reading data from your GCS bucket, you can use the `newInputFile` method.
+
+```java
+InputFile inputFile = gcsFileIO.newInputFile("gs://my_bucket/data/my_data.parquet");
+
+```
+
+Replace `"gs://my_bucket/data/my_data.parquet"` with the path of the data you want to read.
+
+- **Creating an OutputFile**: To write data to your GCS bucket, you would establish an OutputFile using the newOutputFile method.
+
+```java
+OutputFile outputFile = gcsFileIO.newOutputFile("gs://my_bucket/data/my_output.parquet");
+```
+
+Again, replace `"gs://my_bucket/data/my_output.parquet"` with the path where you'd like to write your data.
+
+These steps will allow you to set up GCS as your storage layer for Apache Iceberg and interact with the data stored in GCS using the `GCSFileIO` class.
+
+## Loading Data into Iceberg Tables
+
+### Add Iceberg to Spark environment
+
+To load data into Iceberg tables using Apache Spark, you must first add Iceberg to your Spark environment. It can be done using the `--packages` option when starting the Spark shell or Spark SQL:
+
+```bash
+spark-shell --packages org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:1.3.1
+spark-sql --packages org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:1.3.1
+```
+
+### Configure Spark Catalogs
+
+Catalogs in Iceberg are used to track tables. They can be configured using properties under `spark.sql.catalog.(catalog_name)`. Here is an example of how to configure a catalog:
+
+```bash
+spark-sql --packages org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:1.3.1\
+    --conf spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions \
+    --conf spark.sql.catalog.spark_catalog=org.apache.iceberg.spark.SparkSessionCatalog \
+    --conf spark.sql.catalog.spark_catalog.type=hive \
+    --conf spark.sql.catalog.local=org.apache.iceberg.spark.SparkCatalog \
+    --conf spark.sql.catalog.local.type=hadoop \
+    --conf spark.sql.catalog.local.warehouse=$PWD/warehouse \
+    --conf spark.sql.defaultCatalog=local
```

Comment:
shouldn't this have a `--conf spark.sql.catalog.local.io-impl=org.apache.iceberg.gcp.gcs.GCSFileIO` to properly pick up `GCSFileIO`?

Reply:
`ResolvingFileIO` should [pick it based on the `gs` scheme](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java#L51).

Reply:
+1 `ResolvingFileIO` will automatically select `GCSFileIO` based on the scheme in the next release. It might be better to use the REST catalog as an example as it is easier to set up than a HMS, and no `io-impl` needs to be set.